x-pack/legacy/plugins/infra/public/pages/logs/index.tsx

```diff
@@ -16,72 +16,107 @@ import { HelpCenterContent } from '../../components/help_center_content';
-            defaultMessage: 'Provide feedback for Logs',
-          })}
-        />
+  injectI18n(({ match, intl, uiCapabilities }: LogsPageProps) => {
+    const [sourceId] = useSourceId();
+    const source = useSource({ sourceId });
+    const { hasLogAnalysisCapabilites } = useLogAnalysisCapabilities();
+    return (
+      <Source.Context.Provider value={source}>
+        <ColumnarPage>
+          <DocumentTitle
+            title={intl.formatMessage({
+              id: 'xpack.infra.logs.index.documentTitle',
+              defaultMessage: 'Logs',
+            })}
+          />
 
-        <Header
-          breadcrumbs={[
-            {
-              text: i18n.translate('xpack.infra.header.logsTitle', {
-                defaultMessage: 'Logs',
-              }),
-            },
-          ]}
-          readOnlyBadge={!uiCapabilities.logs.save}
-        />
+          <HelpCenterContent
+            feedbackLink="https://discuss.elastic.co/c/logs"
+            feedbackLinkText={intl.formatMessage({
+              id: 'xpack.infra.logsPage.logsHelpContent.feedbackLinkText',
+              defaultMessage: 'Provide feedback for Logs',
+            })}
+          />
 
-        <AppNavigation>
-          <RoutedTabs
-            tabs={[
-              {
-                title: intl.formatMessage({
-                  id: 'xpack.infra.logs.index.streamTabTitle',
-                  defaultMessage: 'Stream',
-                }),
-                path: `${match.path}/stream`,
-              },
+          <Header
+            breadcrumbs={[
               {
-                title: intl.formatMessage({
-                  id: 'xpack.infra.logs.index.settingsTabTitle',
-                  defaultMessage: 'Settings',
+                text: i18n.translate('xpack.infra.header.logsTitle', {
+                  defaultMessage: 'Logs',
                 }),
-                path: `${match.path}/settings`,
               },
             ]}
+            readOnlyBadge={!uiCapabilities.logs.save}
           />
-        </AppNavigation>
+          {source.isLoadingSource ||
+          (!source.isLoadingSource &&
+            !source.hasFailedLoadingSource &&
+            source.source === undefined) ? (
+            <SourceLoadingPage />
+          ) : source.hasFailedLoadingSource ? (
+            <SourceErrorPage
+              errorMessage={source.loadSourceFailureMessage || ''}
+              retry={source.loadSource}
+            />
+          ) : (
+            <>
+              <AppNavigation>
+                <RoutedTabs
+                  tabs={[
+                    {
+                      title: intl.formatMessage({
+                        id: 'xpack.infra.logs.index.streamTabTitle',
+                        defaultMessage: 'Stream',
+                      }),
+                      path: `${match.path}/stream`,
+                    },
+                    ...(hasLogAnalysisCapabilites
```

Comment:
I feel like using this spread operator/ternary structure to conditionally insert something into the middle of an array is very difficult to read. I'd propose instead to define above the the `return` statement:
```js
const streamTab = {
    title: intl.formatMessage({
       id: 'xpack.infra.logs.index.streamTabTitle',
         ...etc
};
const settingsTab = { ...definition of settings tab }
const analysisTab = { ...definition of analysis tab }

const tabs = hasLogAnalysisCapabilities 
  ? [streamTab, analysisTab, settingsTab] 
  : [streamTab, settingsTab]
```

then pass `tabs` into the `RoutedTabs` prop

Reply:
I'm not particularly attached to either syntax - I've changed this over.